packages/badge/src/css/index.js

```diff
@@ -11,82 +19,87 @@ export default {
     border: `1px solid transparent`,
     borderRadius: '2px',
     display: 'inline-block',
-    fontSize: core.type.fontSizeXSmall,
-    fontWeight: core.type.fontWeightMedium,
-    lineHeight: core.type.lineHeightStandard,
-    padding: `0 ${core.layout.spacingXSmall}`,
+    fontSize: type.fontSizeXSmall,
+    fontWeight: type.fontWeightMedium,
+    lineHeight: type.lineHeightStandard,
+    padding: `0 ${layout.spacingXSmall}`,
     textTransform: 'uppercase'
   },
 
   [defaultWithColor(colors.gray)]: {
-    color: core.colors.white,
-    backgroundColor: core.colors.gray03,
-    borderColor: core.colors.gray03
+    color: colorsTextIcon.highOnDark,
```

Comment:
I think there's a no variation not taken into account here.

Default appearance, color gray, light theme.  

highOnLight should be the text color there.